packages/den/async/promiseTimeout.ts

```diff
@@ -0,0 +1,41 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+/** Error for promise timeouts from `promiseTimeout()` */
+class PromiseTimeoutError extends Error {
+  public constructor(message: string) {
+    super(message);
+    this.name = "PromiseTimeoutError";
+  }
```

Comment:
```suggestion
  name = "PromiseTimeoutError";
```
Does this achieve the same thing?